enumerables.rb

```diff
@@ -0,0 +1,136 @@
+module Enumerable
+
+  def my_each
+    return p to_enum(__method__) unless block_given?
+      i = 0
+      while i < self.size
+        yield(self[i])
+        i += 1
+      end
+      return to_enum
+  end
+
+  def my_each_with_index
+      if block_given?
+        i = 0
+        my_each do
+          yield(self[i], i)
+          i += 1
+        end
+      else
+        return to_enum(__method__)
+      end
+    end
+
+  def my_select
+    return p to_enum(__method__) unless block_given?
+      array = []
+      i = 0
+      while i < self.size
+          array << self[i] if yield(self[i])
+          i +=1                
+      end
+      array
+  end
+
+  def my_all?(pattern = nil) 
+      method_all = true
+      if block_given? 
+        my_each do |i| 
+          method_all = false unless yield i 
+        end
+        elsif pattern
+        my_each do |i| 
+          method_all = false unless pattern === i 
+        end
+      else
+        my_each do |i|
+          method_all = false unless i
+        end
+      end
+      method_all
+    end
+
+  def my_any?(pattern = nil) 
+      method_any = false
+      if block_given? 
+          my_each do |element|
+            method_any = true if yield element 
+          end
+      elsif pattern
+          my_each do |element|
+            method_any = true if pattern === element
+          end
+      else
+          my_each do |element| 
+            method_any = true if element
+          end
+      end
+      method_any
+  end
+
+  def my_none?(pattern = nil)
+      method_none = true
+      if block_given?
+          my_each  do |i| 
+            method_none = false if yield i 
+          end
+      elsif pattern
+          my_each do |i| 
+            method_none = false if pattern === i
+          end
+      else
+          my_each do |i| 
+            method_none = false if i
+          end
+      end
+      method_none
+  end
+
+  def my_count(answer = nil, &block)
+      method_count = 0
+      if block && !answer
+        my_each do |n| 
+          method_count += 1 if yield(n)
+        end
+      elsif !answer
+        my_each do |n|
+          method_count += 1
+        end
+      else
+        my_each do |n| 
+          method_count += 1 if answer == n
+        end
+      end
+      method_count
+    end
+
+    def my_map(&block)
+      if block_given? 
+        result = [] 
+        if block.nil?
+          return result
+        elsif
+          my_each {|item| result << yield(item)}
+        end
+        result
+      else
+        return to_enum(__method__)
+      end
+    end
+
+    def my_inject(*args)
```

Comment:
You have not considered if one of your args is a symbol.
If a symbol is specified combines each element of the collection by applying the symbol as a named method.
You can read more about enumerable methods here: https://ruby-doc.org/core-2.7.0/Enumerable.html#method-i-inject

You may use the [`is_a?`](https://ruby-doc.org/core-2.7.0/Object.html#method-i-is_a-3F) method in the following manner to help you solve this issue:
```ruby
is_a?(Symbol)
```

You may also use the [`send`](https://ruby-doc.org/core-2.7.0/Object.html#method-i-send) method to help handle the symbol.

Hope this helps 🙏

Reply:
Hi Maya, thank you for the feedback, I sent you a message on slack